linuxbrew-build.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+ABS_PATH_LIMIT=85
+export HOMEBREW_NO_AUTO_UPDATE=1
+
+[[ -x ./bin/brew ]] || (echo "This script should be run inside Linuxbrew directory."; exit 1)
+
+cd $(realpath .)
+BREW_HOME="$PWD"
+
+LEN=${#BREW_HOME}
+[[ $LEN -eq $ABS_PATH_LIMIT ]] || (echo "Linuxbrew absolute path should be exactly $ABS_PATH_LIMIT \
+ bytes, but actual length is $LEN bytes: $BREW_HOME"; exit 1)
+
+export HOMEBREW_ARCH=ivybridge
+
+openssl_formula=./Library/Taps/homebrew/homebrew-core/Formula/openssl.rb
+openssl_orig=./Library/Taps/homebrew/homebrew-core/Formula/openssl.rb.orig
+
+if [[ ! -e "$openssl_orig" ]]; then
+  ./bin/brew info openssl >/dev/null
+  cp "$openssl_formula" "$openssl_orig"
+fi
+
+cp "$openssl_orig" "$openssl_formula"
+  cat <<EOF | patch -n $openssl_formula
+4c4
+< class Openssl < Formula
+---
+> class Openssl < Formula
+40c40,41
+<       :x86_64 => %w[linux-x86_64],
+---
+>       :x86_64 => %w[linux-x86_64
+>                     -march=ivybridge -mno-avx -mno-bmi -mno-bmi2 -mno-fma -no-abm -no-movbe],
+EOF
+
+./bin/brew install autoconf automake bzip2 flex gcc icu4c libtool libuuid maven ninja openssl \
+readline s3cmd
+
+if [[ ! -e VERSION_INFO ]]; then
+  commit_id=$(git rev-parse HEAD)
+  echo "Linuxbrew commit ID: $commit_id" >VERSION_INFO.tmp
+  pushd $(pwd)
+  cd Library/Taps/homebrew/homebrew-core
+  commit_id=$(git rev-parse HEAD)
+  popd
+  echo "homebrew-core commit ID: $commit_id" >>VERSION_INFO.tmp
+  mv VERSION_INFO.tmp VERSION_INFO
+fi
+
+echo "Updating symlinks ..."
+cat <<EOF >update_links.sh
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+BREW_HOME="\${0%/*}"
+[[ -x \$BREW_HOME/bin/brew ]] || \
+  (echo "This script should be located inside Linuxbrew directory."; exit 1)
+
+find \$BREW_HOME -type l | while read f
+do
+  target="\$(readlink "\$f")"
+  real_target="\$(realpath "\$f")"
+  if [[ "\$real_target" != "\$BREW_HOME"* && "\$real_target" != "\$target" ]]; then
+    # We want to convert relative links pointing outside of Linuxbrew to absolute links.
+    ln -sfT "\$real_target" "\$f"
+  fi
+done
+EOF
```

Comment:
Why do we need to generate a script on the fly? Why can't we do this as part of the parent script?
An embedded script with escaped characters is difficult to read.